packages/execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -193,9 +200,13 @@ export class BaseSnapExecutor {
     };
 
     this.snapPromiseErrorHandler = (error: PromiseRejectionEvent) => {
-      this.errorHandler('Unhandled promise rejection in snap.', error.reason, {
-        snapName,
-      });
+      this.errorHandler(
+        'Unhandled promise rejection in snap.',
+        error instanceof Error ? error : error.reason,
```

Comment:
lmao, I think this does away with the need for the original logic for `originalError`...me and Erik had came up with this scheme while discussing it....I think we can revert everything back and just keep this line and we would be good right? @rekmarks thoughts?

Reply:
I added this because in Node.js it seems that sometimes `error` is not a promise rejection event.

Reply:
Oh wait you might be right, I think we can revert a lot.